### ISSUE TITLE
Size displaying as "0 x 0" after adding CreateDate with exiftool

### ISSUE THREAD

**Issue by Tyen3** [Webreaper/Damselfly#423]
I'm using the following:
- Damselfly v3.0.0 (Docker install)
- ExifTool v12.45 (latest version)
- Linux 

I have a bunch of JPG images that contain no meta data information. To have them display usefully in Damselfly, I've written a script that modifies each file with exiftool to add the create date. Before modifying the file with exiftool, Damselfly properly shows the image size in the left hand properties pane. After adding the create date, Damselfly shows the correct timestamp in "Date Taken:" as expected, but now shows the "Size:" as 0 x 0.

How to reproduce:
- Download a jpg that contains no image creation date information:
`$ wget https://image-aws-us-west-2.vsco.co/74f24b/257909881/62aa847d265de0282767a2b5/vsco_061522.jpg`
- View the image in Damselfly. "Date Taken:" shows as unknown, "Size:" shows as 640 x 1136
- Get information on the image:
``` 
$ exiftool vsco_061522.jpg
ExifTool Version Number         : 12.45
File Name                       : vsco_061522.jpg
Directory                       : .
File Size                       : 248 kB
File Modification Date/Time     : 2022:06:15 21:16:47-04:00
File Access Date/Time           : 2022:09:30 09:54:27-04:00
File Inode Change Date/Time     : 2022:09:30 09:54:03-04:00
File Permissions                : -rw-rw-r--
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
Image Width                     : 640
Image Height                    : 1136
Encoding Process                : Baseline DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Image Size                      : 640x1136
Megapixels                      : 0.727

```
- Modify the create date tags:
```
$ exiftool  -AllDates="2022:09:30 08:02:43-04:00" -overwrite_original  -m  vsco_061522.jpg
    1 image files updated
```
- Wait for Damselfly to refresh metadata on the image. "Date Taken:" now reflects the date we set with exiftool, 30-Sep-2022 08:02:43, but "Size:" shows 0 x 0.

**Comment by Webreaper** [comment]
Can you try the latest dev built (docker image `webreaper/damselfly:dev`) and try that, please? I've done some fixes for metadata ingestion with that, and fixes some issue around size metadata.

Note that this will be v4.0, which is a massive rewrite to convert the app to Blazor WebAssembly. It's pretty much completely done, but I'm just doing final testing and a few fixes. So you may want to:

1. Copy your damselfly volume folder to something like damselfly_beta
2. Spin up a second image using the new version
3. Test that it all works, including the size metadata fix

before upgrading your 'primary' installation. If you're feeling brave enough to plough ahead on your main instance, please at least ensure you have a backup of the folder with your DB in it!

Let me know how you get on.

**Comment by Webreaper** [comment]
Update: I tested with your image and the exiftool command, and both are showing the correct size for me, so I think v4.0.0 will fix your issue!

**Comment by Tyen3** [comment]
> Update: I tested with your image and the exiftool command, and both are showing the correct size for me, so I think v4.0.0 will fix your issue!

Yes, I performed the same test against v4.0.0 and can confirm that it works! Further, it looks like v4.0.0 uses the file's last modification time for "Date Taken:" when there is no specific EXIF data, which is an awesome improvement as well as it means I don't need to run my script in the first place!

I've encountered a bunch of issues in v4.0.0, which I suspect are because it's in active development. If you'd like me to file PRs for the issues I found I'd be happy to, or I'll just wait 'till the beta/release is out to test again.

Thank you for writing this software, I've been looking for something like this for a long time!

**Comment by Webreaper** [comment]
Great, good to hear. If you could file one issue with "4.0 bugs" and just list them, that'd be good, and I can split them up if they're not already stuff I know about. 👍

**Comment by Webreaper** [comment]
PS, I think I realised why a few things were broken and am pushing a fix, so if you re-pull dev in about 30 mins it should be a lot better.
